src/main/java/jep/ClassList.java

```diff
@@ -159,6 +165,31 @@ private void loadClassPath() {
         }
     }
 
+   /**
+     * Recursively go through a folder and all subdirectories looking for .class
+     * files. Add them all.
+     */
+    private void addClassFilesInTree(File folder, String prefix) {
+        if (!folder.isDirectory()) {
+            throw new IllegalArgumentException("folder is not a Directory");
+        }
+        for (File file : folder.listFiles()) {
+            String entry = file.getName();
+            if (file.isDirectory()) {
+                if (prefix!=null && !prefix.isEmpty()) {
+                    // Include a . between directories only if there was a prefix.
+                    addClassFilesInTree(file, prefix+"."+entry);
+                } else {
+                    // Don't include a prefix--we only care about subdirectories.
+                    addClassFilesInTree(file, entry);
+                }
+            } else if (file.exists() && file.canRead() && entry.toLowerCase().endsWith(".class")) {
+                // We've found a .class file on the file system.  Add it.
+                addClass(prefix, entry.replaceAll(".class$", ""));
```

Comment:
Do you need the $ here?  I don't understand why that's there.

Reply:
That is the regex end of line character. I don't think there is a legal java filename with .class in the middle but it is pretty standard for ensuring you only modify the extension.

Reply:
Thanks.